pkg/plugin/sdk/deployment_source.go

```diff
@@ -111,15 +111,15 @@ func (c *ApplicationConfig[Spec]) parsePluginConfig(pluginName string) error {
 		c.pluginConfigs = nil
 	}()
 
-	if c.pluginConfigs == nil || c.pluginConfigs[pluginName] == nil {
-		// No config is set for this plugin.
-		// Set the Spec to the zero value of the spec type to avoid nil pointer dereference.
-		c.Spec = new(Spec)
-		return nil
+	// It is necessary to prepare config with default value when users doesn't set any config.
```

Comment:
It's not only default config but also custom unmarshalling logic.
The point is to call `json.Unmarshal` to invoke custom unmarshalling logic.

Reply:
@Warashi 
Thank you for the commnet. I fixed the comment. 
Is it ok?
[01b8d51](https://github.com/pipe-cd/pipecd/pull/5937/commits/01b8d517e7322fee785f6b38d9944008d44b06ea)